src/Database/Attach/File.php

```diff
@@ -74,6 +85,18 @@ class File extends Model
     // Constructors
     //
 
+    /**
+     * Constructor
+     */
+    public function __construct(array $attributes = [])
+    {
+        parent::__construct();
+
+        $this->storageDiskKey = $this->getStorageConfigKey() ?
```

Comment:
Why does this need to be a method call instead of just a pre-initialized property? Also, why do we use `Config` here? It may make sense for your specific use case, but it makes the purpose of the key change between being a reference to a config value and being a straight forward reference to an actual storage disk. I think a better option might be to use the method `getStorageDiskKey()` instead of the property altogether and then override this method in the system\models\file to use the config value